jwql/utils/preview_image.py

```diff
@@ -212,13 +212,18 @@ def get_data(self, filename, ext):
                     yd, xd = data.shape[-2:]
                     dq = np.ones((yd, xd), dtype="bool")
 
+
                 # Collect information on aperture location within the
                 # full detector. This is needed for mosaicking NIRCam
                 # detectors later.
-                self.xstart = hdulist[0].header['SUBSTRT1']
-                self.ystart = hdulist[0].header['SUBSTRT2']
-                self.xlen = hdulist[0].header['SUBSIZE1']
-                self.ylen = hdulist[0].header['SUBSIZE2']
+                try:
```

Comment:
So what happens in the case where these keywords are not present? (Is that for level 3 products?) The `xstart` and `ystart` attributes are used in `generate_preview_images`, line 305.

Reply:
@bhilbert4 From what I can gather, these keywords are only missing in level 3 products.  And `xstart` and `ystart` only seem to be used when creating mosaics, so the question arises -- is there a need to create mosaics with level 3 products?

Reply:
I don't think so. At that point mosaics will have already been created for the appropriate data types. So I guess as long as `grouped_files` contains only a single file when it searches for a level 3 file basename, then `create_mosaic` won't be called. I suspect this will be true, but I'm not 100% certain.

Reply:
@bhilbert4 Ok, I think this is the case.  We will go with this for now and deal with any possible bugs later.